pkg/operator/options/options.go

```diff
@@ -0,0 +1,142 @@
+	fs.IntVar(&o.ReservedENIs, "reserved-enis", env.WithDefaultInt("RESERVED_ENIS", 0), "Reserved ENIs are not included in the calculations for max-pods or kube-reserved. This is most often used in the VPC CNI custom networking setup https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html.")
+}
+
+func (o *Options) Parse(fs *flag.FlagSet, args ...string) error {
+	if err := fs.Parse(args); err != nil {
+		if errors.Is(err, flag.ErrHelp) {
+			os.Exit(0)
+		}
+		return fmt.Errorf("parsing flags, %w", err)
+	}
+
+	// Check if each option has been set. This is a little brute force and better options might exist,
+	// but this only needs to be here for one version
+	o.setFlags = map[string]bool{}
+	cliFlags := sets.New[string]()
+	fs.Visit(func(f *flag.Flag) {
+		cliFlags.Insert(f.Name)
+	})
+	fs.VisitAll(func(f *flag.Flag) {
+		envName := strings.ReplaceAll(strings.ToUpper(f.Name), "-", "_")
+		_, ok := os.LookupEnv(envName)
+		o.setFlags[f.Name] = ok || cliFlags.Has(f.Name)
+	})
+
+	if err := o.Validate(); err != nil {
+		return fmt.Errorf("validating options, %w", err)
+	}
+
+	return nil
+}
+
+func (o *Options) ToContext(ctx context.Context) context.Context {
+	return ToContext(ctx, o)
+}
+
+func (o *Options) MergeSettings(ctx context.Context) {
+	s := settings.FromContext(ctx)
+	if s == nil {
+		return
+	}
+
+	mergeField(&o.AssumeRoleARN, s.AssumeRoleARN, o.setFlags["assume-role-arn"])
+	mergeField(&o.AssumeRoleDuration, s.AssumeRoleDuration, o.setFlags["assume-role-duration"])
+	mergeField(&o.ClusterCABundle, s.ClusterCABundle, o.setFlags["cluster-ca-bundle"])
+	mergeField(&o.ClusterName, s.ClusterName, o.setFlags["cluster-name"])
+	mergeField(&o.ClusterEndpoint, s.ClusterEndpoint, o.setFlags["cluster-endpoint"])
+	mergeField(&o.IsolatedVPC, s.IsolatedVPC, o.setFlags["isolated-vpc"])
+	mergeField(&o.VMMemoryOverheadPercent, s.VMMemoryOverheadPercent, o.setFlags["vm-memory-overhead-percent"])
+	mergeField(&o.InterruptionQueueName, s.InterruptionQueueName, o.setFlags["interruption-queue-name"])
+	mergeField(&o.ReservedENIs, s.ReservedENIs, o.setFlags["reserved-enis"])
+
+	if err := o.checkRequiredFields(); err != nil {
+		panic(fmt.Errorf("checking required fields, %w", err))
+	}
+}
+
+func ToContext(ctx context.Context, opts *Options) context.Context {
+	return context.WithValue(ctx, optionsKey{}, opts)
+}
+
+func FromContext(ctx context.Context) *Options {
+	retval := ctx.Value(optionsKey{})
+	if retval == nil {
+		return nil
+	}
+	return retval.(*Options)
+}
+
+// Note: Separated out to help with cyclomatic complexity check
+func mergeField[T any](dest *T, src T, isDestSet bool) {
+	if !isDestSet {
+		*dest = src
+	}
+}
+
+func (o *Options) checkRequiredFields() error {
+	if o.ClusterName == "" {
+		return fmt.Errorf("missing field, cluster-name")
```

Comment:
```suggestion
		return fmt.Errorf("missing required field, cluster-name")
```

Reply:
It's wrapped further up by "checking required fields, %w" so I thought that might be redundant. Thoughts?